deepmd/descriptor/se_a_mask.py

```diff
@@ -0,0 +1,429 @@
+                  seed: int = None,
+                  type_one_side: bool = True,
+                  exclude_types: List[List[int]] = [],
+                  set_davg_zero: bool = True,
+                  activation_function: str = 'tanh',
+                  precision: str = 'default',
+                  uniform_seed: bool = False
+    ) -> None:
+        """
+        Constructor
+        """
+        self.sel_a = sel
+        self.cum_sel_a = np.concatenate((np.array([0]), np.cumsum(self.sel_a)))
+        self.total_atom_num = np.cumsum(self.sel_a)[-1]
+        self.ntypes = len(self.sel_a)
+        self.descrpt_type = "se_a_mask"
+        # rcut and rcut_smth will not be used in se_a_mask op.
+        self.rcut_r = rcut
+        self.rcut_r_smth = rcut_smth
+        self.type_one_side = False
+        self.type_embedding = None
+        
+        self.filter_neuron = neuron
+        self.n_axis_neuron = axis_neuron
+        self.filter_resnet_dt = resnet_dt
+        self.seed = seed
+        self.uniform_seed = uniform_seed
+        self.seed_shift = embedding_net_rand_seed_shift(self.filter_neuron)
+        self.trainable = trainable
+        self.compress_activation_fn = get_activation_func(activation_function)
+        self.filter_activation_fn = get_activation_func(activation_function)
+        self.filter_precision = get_precision(precision)
+        self.exclude_types = set()
+        for tt in exclude_types:
+            assert(len(tt) == 2)
+            self.exclude_types.add((tt[0], tt[1]))
+            self.exclude_types.add((tt[1], tt[0]))
+        self.set_davg_zero = set_davg_zero
+
+        # descrpt config
+
+        # numb of neighbors and numb of descrptors
+        self.nnei_a = np.cumsum(self.sel_a)[-1]
+        self.nnei = self.nnei_a
+
+        self.ndescrpt_a = self.nnei_a * 4
+        self.ndescrpt = self.ndescrpt_a
+        self.useBN = False
+        self.dstd = None
+        self.davg = None
+        self.compress = False
+        self.embedding_net_variables = None
+        self.mixed_prec = None
+        self.place_holders = {}
+        nei_type = np.array([])
+        for ii in range(self.ntypes):
+            nei_type = np.append(nei_type, ii * np.ones(self.sel_a[ii])) # like a mask 
+        self.nei_type = tf.constant(nei_type, dtype = tf.int32)
+
+        avg_zero = np.zeros([self.ntypes,self.ndescrpt]).astype(GLOBAL_NP_FLOAT_PRECISION)
+        std_ones = np.ones ([self.ntypes,self.ndescrpt]).astype(GLOBAL_NP_FLOAT_PRECISION)
+        sub_graph = tf.Graph()
+        with sub_graph.as_default():
+            name_pfx = 'd_sea_mask_'
+            for ii in ['coord']:
+                self.place_holders[ii] = tf.placeholder(GLOBAL_NP_FLOAT_PRECISION, [None, None], name = name_pfx+'t_'+ii)
+            self.place_holders['type'] = tf.placeholder(tf.int32, [None, None], name=name_pfx+'t_type')
+            self.place_holders['mask_matrix'] = tf.placeholder(tf.int32, [None, None], name=name_pfx+'t_mask')
```

Comment:
From the docs I see `mask_matrix` can be either `0` or `1`, so why not use `tf.bool`?

Reply:
That's a legacy code. `tf.bool` might be ok. But I haven't test it. I will try to change this into `tf.bool` in next commit.